classes/data/Transfer.class.php

```diff
@@ -690,6 +704,14 @@ public function __get($property) {
             $recipients = array_values($this->recipients);
             return $recipients[0]->download_link;
         }
+        if($property == 'salt') {
+            if( strlen($this->salt)) {
+                return $this->salt;
+            }
+            $this->salt = substr(Utilities::generateRandomHexString(),0,32);
```

Comment:
In effect you are *halving* your salt length here. 16^32 = 2^128 so you actually have only a salt of 128 bits. Which is still adequate, but this is a bad idea because it is very confusing! 

How to properly generate a random hex string in PHP:

```ParagonIE\ConstantTime\Hex::encode(random_bytes(32));```

In JS you would reverse this by 'hex decoding', **in constant time** this "salt" and use it in the key derivation algorithm.

Also please note that it would be much better to use Argon2 for deriving a key from a password in this day and age.